### PR TITLE
{data}[gompi/2024a] netCDF-C++4 v4.3.1, netCDF-Fortran v4.6.1, netCDF v4.9.2, ...

### PR DESCRIPTION
**Pull request by maxim-masterov**
(created using `eb --new-pr`)

### CODE REVIEW

File: easybuild/easyconfigs/n/netCDF-C++4/netCDF-C++4-4.3.1-gompi-2024a.eb [hunk 1]
```diff
@@ -0,0 +1,28 @@
...
+sources = ['v%(version)s.tar.gz']
+checksums = ['e3fe3d2ec06c1c2772555bf1208d220aab5fee186d04bd265219b0bc7a978edc']
+
+dependencies = [
+    ('netCDF', '4.9.2'),
+]
+
+
+sanity_check_paths = {
+    'files': ['include/netcdf', 'lib/libnetcdf_c++4.a', 'lib/libnetcdf_c++4.so'],
```
Comment (smoors) [inline easybuild/easyconfigs/n/netCDF-C++4/netCDF-C++4-4.3.1-gompi-2024a.eb:24]:
```suggestion
    'files': ['include/netcdf', 'lib/libnetcdf_c++4.a', 'lib/libnetcdf_c++4.%s' % SHLIB_EXT],
```

**Comment by smoors** [comment]
@boegelbot: please test @ generoso

**Comment by maxim-masterov** [comment]
The `nc_test4_run_par_test ` test failed. I think it might be caused by an insufficient number of cores: https://github.com/Unidata/netcdf-c/issues/2810#issuecomment-2458803958

Indeed, this test requires 16 cores, see here: https://github.com/Unidata/netcdf-c/blob/v4.9.2/nc_test4/run_par_test.sh.in#L23

**Comment by SebastianAchilles** [comment]
@boegelbot please test @ jsc-zen3
CORE_CNT=16

**Comment by SebastianAchilles** [comment]
@boegelbot please test @ generoso
CORE_CNT=16

**Comment by SebastianAchilles** [comment]
@boegelbot please test @ jsc-zen3

**Comment by SebastianAchilles** [comment]
> > The `nc_test4_run_par_test ` test failed. I think it might be caused by an insufficient number of cores: [Unidata/netcdf-c#2810 (comment)](https://github.com/Unidata/netcdf-c/issues/2810#issuecomment-2458803958)
> > Indeed, this test requires 16 cores, see here: https://github.com/Unidata/netcdf-c/blob/v4.9.2/nc_test4/run_par_test.sh.in#L23
> 
> that should have been worked around by the `pretestopts` in the easyconfig. not sure why that is not working here though, especially since the exact same version worked in the previous toolchain #19534
> 
> ```python
> # some tests try to start 16 MPI ranks, so we need to allow oversubscription to avoid failing tests
> pretestopts = "OMPI_MCA_rmaps_base_oversubscribe=1 "
> ```

We are using OpenMPI v5.0.3 in the 2024a toolchain generation. OpenMPI v5.0 has moved to PRRTE as its runtime, therefore `OMPI_MCA_rmaps_base_oversubscribe` won't work anymore, we could consider setting the following instead:
```
PRTE_MCA_rmaps_base_oversubscribe=1
```

File: easybuild/easyconfigs/n/netCDF/netCDF-4.9.2-gompi-2024a.eb [hunk 2]
```diff
@@ -0,0 +1,54 @@
...
+
+# make sure both static and shared libs are built
+# and disable "remote" tests that access a unreliable external test server over internet
+configopts = [
+    "-DENABLE_DAP_REMOTE_TESTS=OFF -DBUILD_SHARED_LIBS=OFF",
+    "-DENABLE_DAP_REMOTE_TESTS=OFF -DBUILD_SHARED_LIBS=ON",
+]
+
+# some tests try to start 16 MPI ranks, so we need to allow oversubscription to avoid failing tests
+pretestopts = "OMPI_MCA_rmaps_base_oversubscribe=1 "
```
Comment (SebastianAchilles) [inline easybuild/easyconfigs/n/netCDF/netCDF-4.9.2-gompi-2024a.eb:51]:
```suggestion
pretestopts = "PRTE_MCA_rmaps_default_mapping_policy=:oversubscribe "
```

**Review by SebastianAchilles — DISMISSED** [review]

**Comment by SebastianAchilles** [comment]
@boegelbot please test @ jsc-zen3

**Review by smoors — APPROVED** [review]
lgtm

**Comment by smoors** [comment]
Going in, thanks @maxim-masterov!

**Pull request merged**